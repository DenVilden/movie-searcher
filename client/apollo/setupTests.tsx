import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
