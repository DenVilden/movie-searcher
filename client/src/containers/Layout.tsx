import React from 'react';
import Header from './Header';

type Props = {
  children: React.ReactElement;
};

const Layout = ({ children }: Props) => (
  <>
    <Header />
    {children}
  </>
);

export default Layout;
