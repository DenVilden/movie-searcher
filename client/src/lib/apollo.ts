import { ApolloClient, InMemoryCache, HttpLink, makeVar } from "@apollo/client";
import { withApollo } from "next-apollo";

if (process.env.NODE_ENV === "test") {
  // eslint-disable-next-line global-require
  require("cross-fetch/polyfill");
}

export const favoritesVar = makeVar<string[]>([]);

const cache = new InMemoryCache();

export const client = new ApolloClient({
  link: new HttpLink({
    uri: process.env.NEXT_PUBLIC_SERVER_URL,
    credentials: "same-origin",
  }),
  cache,
});

export default withApollo(client)({ ssr: true });
