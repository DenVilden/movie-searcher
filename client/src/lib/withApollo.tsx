import withApollo from 'next-with-apollo';
import { ApolloClient, InMemoryCache, gql } from '@apollo/client';
import { loader } from 'graphql.macro';
import { getDataFromTree } from '@apollo/react-ssr';
import resolvers from '../graphql/resolvers';

const createClient = ({ initialState }: any) => {
  const client = new ApolloClient({
    typeDefs: loader('../graphql/schema.graphql'),
    resolvers,
    uri: process.env.SERVER_URL,
    cache: new InMemoryCache().restore(initialState || {}),
  });
  client.writeQuery({
    query: gql`
      {
        favorites
        inputValue
      }
    `,
    data: { favorites: [], inputValue: '' },
  });
  return client;
};

export default withApollo(createClient, { getDataFromTree });
