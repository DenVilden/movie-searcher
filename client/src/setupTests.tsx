import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
