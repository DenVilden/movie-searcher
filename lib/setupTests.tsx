/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
