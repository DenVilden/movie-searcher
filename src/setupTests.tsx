import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MockedProvider, MockedResponse } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import { createMuiTheme } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

const theme = createMuiTheme();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>
        <Router>{element}</Router>
      </ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
