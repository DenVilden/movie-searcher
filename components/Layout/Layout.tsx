import { StylesProvider, CssBaseline } from '@material-ui/core';
import { ThemeProvider, createGlobalStyle } from 'styled-components';
import Head from 'next/head';
import { Header } from '..';
import { theme } from '../../lib/theme';

const GlobalStyle = createGlobalStyle`
  body {
    min-width: 320px;
  }

  @font-face {
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-display: swap;
    src: url('/fonts/Roboto-Regular.woff2');
  }
  @font-face {
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-display: swap;
    src: url('/fonts/Roboto-Medium.woff2');
  }
  @font-face {
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 700;
    font-display: swap;
    src: url('/fonts/Roboto-Bold.woff2');
  }
`;

interface Props {
  children: React.ReactElement;
}

export default function Layout({ children }: Props) {
  return (
    <>
      <Head>
        <link
          rel="preload"
          href="/fonts/Roboto-Regular.woff2"
          as="font"
          crossOrigin=""
        />
        <link
          rel="preload"
          href="/fonts/Roboto-Medium.woff2"
          as="font"
          crossOrigin=""
        />
        <link
          rel="preload"
          href="/fonts/Roboto-Bold.woff2"
          as="font"
          crossOrigin=""
        />
        <link href="/favicon.ico" rel="shortcut icon" />
        <meta charSet="utf-8" />
        <meta content="width=device-width, initial-scale=1" name="viewport" />
        <meta content="Movie searcher app" name="description" />
        <title key="title">Movie Searcher</title>
      </Head>
      <StylesProvider injectFirst>
        <ThemeProvider theme={theme}>
          <CssBaseline />
          <GlobalStyle />
          <Header />
          {children}
        </ThemeProvider>
      </StylesProvider>
    </>
  );
}
