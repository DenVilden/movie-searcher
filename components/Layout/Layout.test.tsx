import { render } from '../../lib/setupTests';
import Layout from './Layout';

describe('layout', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Layout>
        <div />
      </Layout>,
    );

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
