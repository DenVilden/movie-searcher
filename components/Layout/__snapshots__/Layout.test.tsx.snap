// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular Header__StyledToolbar-sc-1j8nxqe-0 cMdcLq MuiToolbar-gutters"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Header__StyledMovieIcon-sc-1j8nxqe-1 kCKuMD"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 4l2 3h-3l-2-3h-2l2 3h-3l-2-3H8l2 3H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4zm-6.75 11.25L10 18l-1.25-2.75L6 14l2.75-1.25L10 10l1.25 2.75L14 14l-2.75 1.25zm5.69-3.31L16 14l-.94-2.06L13 11l2.06-.94L16 8l.94 2.06L19 11l-2.06.94z"
        />
      </svg>
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root SearchBar__StyledAutocomplete-gyywkg-0 jGgSbd MuiAutocomplete-hasClearIcon"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root SearchBar__StyledInputBase-gyywkg-1 hreRla MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="autocomplete"
              placeholder="search for movies and tv shows"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-2"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root Favorites__StyledIconButton-w5bsin-0 fHDBVm MuiIconButton-colorInherit Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
            >
              0
            </span>
          </span>
        </span>
      </button>
    </div>
  </header>
  <div />
</DocumentFragment>
`;
